lib/pippi/checks/method_sequence_checker.rb

```diff
@@ -38,12 +46,17 @@ def decorate
       end
 
       # e.g., "select" in "select followed by size"
+      #
+      # Create a new module which will be used to intercept method calls via Module#prepend
       first_method_decorator = Module.new do
         descriptor = method_sequence_check_instance.check_descriptor.method_sequence.method1
+        # Intercept and decorate the original method call
+        # Dynamically add the second decorator to the return value via Object#extend
         define_method(descriptor) do |*args, &blk|
           result = super(*args, &blk)
           if self.class.instance_variable_get(name)
             result.extend second_method_decorator
+            # TODO - @tcopeland could you explain this block?
```

Comment:
@jbodah this one is to handle this case:

```
x = foo.select {|y| y > 2 }
x.sort!
z = x.first
```

That is, in this case there's a mutator method that's called between `select` and `first`, and thus that method sequence can't be replaced with `detect`.